FWCore/Framework/interface/ESProducer.h

```diff
@@ -242,6 +241,105 @@ namespace edm {
       return ESConsumesCollectorT<TRecord>(consumesInfos_.back().get(), id);
     }
 
+    // These replicate the setWhatProduced functions but add a second functor for
+    // the acquire step. The acquire step is analogous to the acquire step for
+    // EDProducers with the ExternalWork ability.
+
+    template <typename T>
+    auto setWhatProducedExternalWork(T* iThis, const es::Label& iLabel = {}) {
```

Comment:
I had initially thought something along `setWhatProducedAsync()`, but maybe the common name with the ED-side module ability is better. @Dr15Jones, what do you think?

Reply:
I don't feel strongly either way. I think both Async or ExternalWork are good suffixes for this. Note that I made the two function names using the lambda argument different because I was getting ambiguity errors when compiling with them. The compiler couldn't resolve the overloaded functions.

Reply:
Would this be better as `setWhatAcquiredProduced` since that matches the naming convention better?

Reply:
Rather than add this to ESProducer, how about we make a new ESExteralWorkProducer or ESAcquirerProducer?

Reply:
> Rather than add this to ESProducer, how about we make a new ESExteralWorkProducer or ESAcquirerProducer?

On this point, just to raise a question, would it make sense for one ESProducer to (be able to) have both regular `produce()` and asynchronous `acquire()`+`produce()` functions? (from a technical point of view it is possible, and the approach in this PR allows it)

Reply:
I think it would be good to allow both.